bindings/nodejs/lib/types/block/id.ts

```diff
@@ -21,13 +43,18 @@ export type NftId = HexEncodedString;
 /**
  * A Block ID represented as hex-encoded string.
  */
-export type BlockId = HexEncodedString;
+export class BlockId extends IdWithSlotIndex {}
```

Comment:
Here we could get away with `BlockId` still being a type
`type BlockId = IdWithSlotIndex` if that's better 🤔 
You won't be able to instantiate it as a class, but you should still be able to do:
```
const blockId: BlockId = new IdWithSlotIndex("12345678")
```
Again, not sure if it's better, just cleaner maybe.

Reply:
Or maybe just rename `IdWithSlotIndex` to `BlockId`.

```tsx
  const slot = BlockId("0x...").slotIndex();
```

Reply:
But then you can't call `.slotIndex()` on a blockid?

Reply:
@Thoralf-M In which case you mean?
For the example i suggested up here you can do.
```
const blockId: BlockId = new IdWithSlotIndex("12345678")
blockId.slotIndex()
```
Cause the `BlockId` type is just an alias for `IdWithSlotIndex`

Reply:
Now I understand, thanks
So do you prefer this, or https://github.com/iotaledger/iota-sdk/pull/1962#discussion_r1481491384? :smile:

Reply:
@msarcev You know, at this point I think the solution @Thoralf-M  did previously is not looking bad to me. Because in case another util method is added to the `class IdWithSlotIndex extends String`, for example `isValid()`, all the other classes that extend from have it, also have that util available, like OutputId. This way creating and referencing the util function multiple times in different classes.

TLDR; to me like it is done right now is fine, as they will be types correctly without needing an extra type
```tsx
export class BlockId extends IdWithSlotIndex {}
export class TransactionId extends IdWithSlotIndex {}
// ...
```

Reply:
Agreed, we can go with this imo.